webapp/src/main/java/com/box/l10n/mojito/okapi/filters/AndroidXMLEncoder.java

```diff
@@ -2,41 +2,56 @@
 
 import net.sf.okapi.common.encoder.EncoderContext;
 import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Configurable;
 
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 /**
  * This overrides the {@link net.sf.okapi.common.encoder.XMLEncoder} for Android
  * strings.
- *
+ * <p>
  * It does not escape supported HTML elements for Android strings unless there
  * are variables within the HTML elements.
  * For example, <b>songs</b> vs. &lt;b>%d songs&lt;/b>
- *
+ * <p>
  * Also it overrides the default quotemode setting so the quotes do not get escaped.
- *
+ * <p>
  * For detailed information, see to Android specification in
  * http://developer.android.com/guide/topics/resources/string-resource.html,
  *
  * @author jyi
  */
+@Configurable
 public class AndroidXMLEncoder extends net.sf.okapi.common.encoder.XMLEncoder {
 
+    /**
+     * logger
+     */
+    static Logger logger = LoggerFactory.getLogger(AndroidXMLEncoder.class);
+
     // trying to match variables between html tags, for example, <b>%d</b>, <i>%1$s</i>, <u>%2$s</u>
-    private static final Pattern ANDROID_VARIABLE_WITHIN_HTML = Pattern.compile("(&lt;[b|i|u]&gt;)((.*?)%(([-0+ #]?)[-0+ #]?)((\\d\\$)?)(([\\d\\*]*)(\\.[\\d\\*]*)?)[dioxXucsfeEgGpn](.*?))+(&lt;/[b|i|u]&gt;)");
-    private static final Pattern ANDROID_HTML = Pattern.compile("(&lt;)(/?)(b|i|u)(&gt;)");
+    private static final Pattern ANDROID_VARIABLE_WITHIN_HTML = Pattern.compile("(&lt;(?:b|i|u|annotation.*?)&gt;)((.*?)%(([-0+ #]?)[-0+ #]?)((\\d\\$)?)(([\\d\\*]*)(\\.[\\d\\*]*)?)[dioxXucsfeEgGpn](.*?))+(&lt;/(?:b|i|u|annotation.*?)&gt;)");
+    private static final Pattern ANDROID_HTML = Pattern.compile("(&lt;)(/?)(b|i|u|annotation.*?)(&gt;)");
     private static final Pattern UNESCAPED_DOUBLE_QUOTE = Pattern.compile("([^\\\\])(\")");
     private static final Pattern START_WITH_DOUBLE_QUOTE = Pattern.compile("(^\")");
     private static final Pattern UNESCAPED_SINGLE_QUOTE = Pattern.compile("([^\\\\])(')");
     private static final Pattern START_WITH_SINGLE_QUOTE = Pattern.compile("(^')");
+    private static final Pattern LINE_FEED = Pattern.compile("\n");
+    private static final Pattern CARIAGE_RETURN = Pattern.compile("\r");
 
     /**
      * New escaping, should be come default but keep it as an option for backward compatibility. Can invert the option
      * later to old escaping or just remove it
      */
     boolean newEscaping = false;
 
+    @Autowired
+    UnescapeUtils unescapeUtils;
+
     public AndroidXMLEncoder(boolean newEscaping) {
         this.newEscaping = newEscaping;
```

Comment:
@jeeyi If you get a chance to test the new escaping, I think it will be nice to turn it on by default.

to use it right now: `mojito push -r android -fo newEscaping=true`

Reply:
Let me test it and get back to you

Reply:
@aurambaj With the new escaping, some strings need re-translation because they do not match anymore. I am okay to make the new escaping the default as long as there is a way to use the old one. I can update our push command for Android.

Reply:
Yes that make sense since the behavior is slightly different. I believe that accepting the old string should be fine since it should be backward compatible in term of Android app but I haven't done extensive testing. 

I'm gonna swap the option for `oldEscaping` then. Usage will be `-fo oldEscaping=true` and if not set it will use the new escaping by default

Reply:
Got it!

Reply:
Also I think most of the change should be backward compatible (only changing how quotes are handled). The things that might be different are line feed. They used to show escaped in the translation and now, it is real line feed. Are those the main differences you see?

I'm thinking a good way could be to just push+import for the current files with the new escaping. It will re-import the small diff and then you don't have to use `-fo oldEscaping=true` at all.